src/hotspot/share/runtime/os.hpp

```diff
@@ -644,7 +645,8 @@ class os: AllStatic {
   //File i/o operations
 
   static ssize_t read_at(int fd, void *buf, unsigned int nBytes, jlong offset);
-  static ssize_t write(int fd, const void *buf, unsigned int nBytes);
+  // Writes the bytes completely. Returns 0 on success, -1 otherwise.
+  static ssize_t write(int fd, const void *buf, size_t nBytes);
```

Comment:
We don't need a ssize_t return type if we only return -1 or 0. Also this should be specified to return OS_ERR on error and OS_OK on success. The callers should also check for OS_ERR and OS_OK rather than -1, < 0, > 0 etc.

Or we could simply make this a boolean function.

Reply:
Return value changed to `bool`. All calls changed accordingly.